src/discljord/messaging/impl.clj

```diff
@@ -111,7 +111,8 @@
        ~params [] ~opts ~method ~status ~body
        ~url
        ~(if delete? `{} `{:body (json/write-str ~opts)})
-       ~(if delete? `(= ~status 204) `(json-body ~body)))))
+       ~(if delete? `(= ~status 204) `(cond->> (json-body ~body)
+                                        (not= 2 (quot ~status 100)) (ex-info (str "Attempted to " ~name " with invalid parameters")))))))
```

Comment:
This condition (`(not= 2 (quot ~status 100))`) should really be checked in `defdispatch` so that it applies to (almost) all dispatch definitions, and not just message updates. The message also should rather be something like 
`(str "Received an error response from " name " (status " status ")")` - there are more reasons for failure than just invalid parameters and the status should be mentioned.

This change requires a bit more effort because:
- existing `ex-info` wrappings like in `bulk-overwrite-global-application-commands!` need to be removed
- not every dispatch implementation is written using `defdispatch`, although I think the only separate one to look out for is the `send-message!` private function.

If you don't have time to make this change, just say the word and I'll make it myself. But as it stands, this only fixes a part of the inconsistency, ideally all of it should be fixed.

Reply:
Makes sense. Just pushed, I'm giving it a test now.